src/main/kotlin/eu/gillissen/rider/usersecrets/InitUserSecretsAction.kt

```diff
@@ -0,0 +1,72 @@
+package eu.gillissen.rider.usersecrets
+
+import com.intellij.notification.NotificationGroupManager
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.vfs.VirtualFile
+
+class InitUserSecretsAction : AnAction() {
+    override fun update(actionEvent: AnActionEvent) {
+        actionEvent.presentation.isEnabledAndVisible = false
+
+        if (!isActionSupported(actionEvent)) {
+            return
+        }
+
+        actionEvent.presentation.isVisible = true
+
+        // If Project already contains <UserSecretsId>
+        if (!actionEvent.getXmlUserSecretsIdValue().isNullOrEmpty()) {
+            return
+        }
```

Comment:
I don't think it makes sense to show the action if the project already has user secrets.

```suggestion
        if (!isActionSupported(actionEvent) || !actionEvent.getXmlUserSecretsIdValue().isNullOrEmpty()) {
            return
        }
```

Reply:
Or better yet, add it to the `isActionSupported` method.

Reply:
The idea behind disabled but visible is just about verbosity. In such a case the user will know that he has such ability to initialize secrets, but now for the selected project. With both disabled + not visible in such a case there is no way to check if the plugin not working or the project is already configured (scenario: user looks at a project the first time and doesn't know if secrets are already installed).

`isActionSupported` function only checks that the selected item is a suitable .NET project file, it is not about the whole validation flow. Action will be visible, but not active until the next check will confirm that the project file is suitable to introduce new secrets id.

But if you prefer not to go with such verbose flow, I will change it to how it worked before.

Reply:
I don't agree with the reasoning about being able to check if its working, as the same code is used for the other action.
The scenario you described is also not relevant as the other action will be visible when the project supports user secrets, and will be disabled when its not yet configured.

On the other hand, I DO agree with showing the disabled action to inform the user of its existence. So I accept the current proposed solution to show both actions when its a project file, and disable the open action when there are no secrets, and the initialize action when they are added.